primitives/mining/staking/src/lib.rs

```diff
@@ -34,7 +37,20 @@ pub trait AssetMining<Balance> {
     }
 }
 
-impl<Balance> AssetMining<Balance> for () {
+#[impl_for_tuples(1, 5)]
```

Comment:
```suggestion
#[impl_for_tuples(5)]
```

So that you can remove the manual impl for `()`.